### PR TITLE
frigate: init at 0.12.0

### PR DESCRIPTION
**Pull request by mweinelt**
###### Description of changes

WIP frigate nvr software
https://github.com/blakeblackshear/frigate

Would be ideal if libedgetpu/pycoral were packaged, but I'm unlikely to do that. #188719

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
````
openvino> -- /nix/store/v73ppkjk196bwsjm6yy9yv80bdq7q68v-pkg-config-wrapper-0.29.2/bin/pkg-config: tbb (2020.3) is found at /nix/store/2d2lhp3b4fwva16v4hnxswwc99z0k3ab-tbb-2020.3
openvino> -- TBB (2020.3) is found at 
openvino> -- Downloading from https://download.01.org/opencv/master/openvinotoolkit/thirdparty/linux/tbbbind_2_5_static_lin_v2.tgz to /build/source/temp/download/tbbbind_2_5_static_lin_v2.tgz ...
openvino> -- Downloading from https://download.01.org/opencv/master/openvinotoolkit/thirdparty/linux/tbbbind_2_5_static_lin_v2.tgz to /build/source/temp/download/tbbbind_2_5_static_lin_v2.tgz ...
openvino> CMake Error at cmake/developer_package/download/download.cmake:9 (file):
openvino>   file DOWNLOAD HASH mismatch
openvino> 
openvino>     for file: [/build/source/temp/download/tbbbind_2_5_static_lin_v2.tgz]
openvino>       expected hash: [865e7894c58402233caf0d1b288056e0e6ab2bf7c9d00c9dc60561c484bc90f4]
openvino>         actual hash: [e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855]
openvino>              status: [6;"Couldn't resolve host name"]
```

Found tbb via pkg-config, but still tries to download it. Maybe a version constraint. Maybe tomorrow.

**Comment by mweinelt** [comment]
```
openvino> /nix/store/r2b9k28c6aghczpqfvh71y9xavm7rr68-binutils-2.39/bin/ld: /nix/store/2d2lhp3b4fwva16v4hnxswwc99z0k3ab-tbb-2020.3/lib/libtbb.so.2: undefined reference to `std::__exception_ptr::exception_ptr::_M_release()@CXXABI_1.3.13'
openvino> /nix/store/r2b9k28c6aghczpqfvh71y9xavm7rr68-binutils-2.39/bin/ld: /nix/store/2d2lhp3b4fwva16v4hnxswwc99z0k3ab-tbb-2020.3/lib/libtbb.so.2: undefined reference to `std::__exception_ptr::exception_ptr::_M_addref()@CXXABI_1.3.13'
openvino> collect2: error: ld returned 1 exit status
openvino> make[2]: *** [docs/snippets/CMakeFiles/ov_integration_snippet.dir/build.make:98: bin/intel64/Release/ov_integration_snippet] Error 1
openvino> make[1]: *** [CMakeFiles/Makefile2:5351: docs/snippets/CMakeFiles/ov_integration_snippet.dir/all] Error 2
openvino> make[1]: *** Waiting for unfinished jobs....
````

**Comment by mweinelt** [comment]
When trying to build with `-DWITH_SYSTEMPROTOBUF:BOOL=ON` I'm getting

```
openvino> -- Found Protobuf: /nix/store/qnnpqln7cx726mfnx8348yil8jbk4l2k-protobuf-3.21.12/lib/libprotobuf.so (found suitable version "3.21.12", minimum required is "3.18.2") 
openvino> CMake Error at thirdparty/CMakeLists.txt:261 (set_property):
openvino>   set_property could not find TARGET protobuf::libprotobuf.  Perhaps it has
openvino>   not yet been created.
openvino> 
openvino> 
openvino> CMake Error at thirdparty/CMakeLists.txt:261 (set_property):
openvino>   set_property could not find TARGET protobuf::libprotobuf-lite.  Perhaps it
openvino>   has not yet been created.
openvino> 
openvino> 
openvino> CMake Error at thirdparty/CMakeLists.txt:282 (target_compile_options):
openvino>   Cannot specify compile options for target "protobuf::libprotobuf-lite"
openvino>   which is not built by this project.
openvino> 
openvino> 
openvino> CMake Error at thirdparty/CMakeLists.txt:282 (target_compile_options):
openvino>   Cannot specify compile options for target "protobuf::libprotobuf" which is
openvino>   not built by this project.
```

**Comment by cpcloud** [comment]
I can try to revive my local libedgetpu branch if that's useful.

**Comment by mweinelt** [comment]
> I can try to revive my local libedgetpu branch if that's useful.

Yeah, I think that would be cool for completeness. Though with the upcoming openvino support it is unlikely that I will get a coral accelerator.

**Comment by mweinelt** [comment]
Config file via environment variable:

https://github.com/blakeblackshear/frigate/blob/dev/frigate/app.py#LL73C23-L73C23

**Comment by mweinelt** [comment]
nginx config
https://github.com/blakeblackshear/frigate/blob/v0.12.0-rc2/docker/rootfs/usr/local/nginx/conf/nginx.conf

**Comment by mweinelt** [comment]
Added an initial version for the nginx config. The tests can now run frigate, and it fails to find the camera, that doesn't exist. Would be cool if we could find something to mock the rtmp stream, if anyone knows anything in that field.

File: pkgs/applications/video/frigate/default.nix [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    hash = "sha256-kLszpjTgQZFMwYGapd+ZgY5sOWxNLblSwP16nP/Eck8=";
+  };
+  tflite_edgetpu_model = fetchurl {
+    url = "https://github.com/google-coral/test_data/raw/release-frogfish/ssdlite_mobiledet_coco_qat_postprocess_edgetpu.tflite";
+    hash = "sha256-Siviu7YU5XbVbcuRT6UnUr8PE0EVEnENNV2X+qGzVkE=";
+  };
+
+  # OpenVino models
+  # https://github.com/blakeblackshear/frigate/blob/v0.12.0/Dockerfile#L92-L95
+  # TODO
```
Comment (mweinelt) [inline pkgs/applications/video/frigate/default.nix:67]: todo

**Comment by mweinelt** [comment]
What works:
- RTSP Streaming
- Storage View
- System View

What does not work:
- Logs
  - expects file based logs, we use systemd-journald, process is not allowed to query the journal

Untested:
- Events
- Birdeye View
  - saw it stream data over a websockets, but nothing happened; possibly related to events

**Review by 0x4A6F — APPROVED** [review]

**Pull request merged**